packages/web-client/app/services/card-customization-options.ts

```diff
@@ -0,0 +1,91 @@
+import Service from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import config from '../config/environment';
+import {
+  ColorCustomizationOption,
+  PatternCustomizationOption,
+} from '../utils/web3-strategies/types';
+
+interface ColorCustomizationResponseObject {
+  attributes: {
+    'pattern-color': string;
+    'text-color': string;
+    background: string;
+    description?: string;
+  };
+  id: string;
+}
+
+interface PatternCustomizationResponseObject {
+  attributes: {
+    'pattern-url': string;
+    description?: string;
+  };
+  id: string;
+}
+
+let convertToColorCustomizationOption = (
+  o: ColorCustomizationResponseObject
+): ColorCustomizationOption => {
+  return {
+    patternColor: o.attributes['pattern-color'],
+    textColor: o.attributes['text-color'],
+    headerBackground: o.attributes.background,
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
+
+let convertToPatternCustomizationOption = (
+  o: PatternCustomizationResponseObject
+): PatternCustomizationOption => {
+  return {
+    patternUrl: o.attributes['pattern-url'],
+    description: o.attributes.description,
+    id: o.id,
+  };
+};
+
+export default class CardCustomizationOptions extends Service {
+  @tracked loaded = false;
+  @tracked patternOptions: PatternCustomizationOption[] | null = null;
+  @tracked colorOptions: ColorCustomizationOption[] | null = null;
+
+  constructor(props: any) {
+    super(props);
+    this.fetchCustomizationOptions();
+  }
+
+  async fetchCustomizationOptions() {
```

Comment:
Two thoughts here:

1) Consider making this an ember-concurrency task. The reason is that it will join the asynchrony with the lifecycle of the service so that in tests where the service is instantiated but end quickly, we don't need to worry about promise resolutions trying to set properties on a service instance or doing pointless parsing.
2) Consider initiating this operation in response to a method call rather than instantiation. It seems cleaner to me to do this rather than couple creation of the service and initiation of this operation.

I feel more strongly about 1 than 2. Both are judgement calls.

Reply:
Does 2. mean that any component that uses this service will need to call the method to fetch the customizations?

Reply:
Situation where this might be relevant is displaying single prepaid cards on the dashboard/in arbitrary threads. Originally was going to handle this by providing a map for ids to styles in this service.

Reply:
My understanding of initiating in response to a method call is in https://github.com/cardstack/cardstack/pull/1732

Reply:
update after discussion w @lukemelia - we are fetching details about styles directly, and we don't need the customization options service to be able to convert ids to styles. This means the only consumer of this service is the form to edit customization options, for now. Will move the customization fetching to a method.

Reply:
This is done!